src/NzbDrone.Core/Music/AddAlbumService.cs

```diff
@@ -20,40 +21,93 @@ public interface IAddAlbumService
     public class AddAlbumService : IAddAlbumService
     {
         private readonly IAlbumService _albumService;
+        private readonly IReleaseService _releaseService;
         private readonly IProvideAlbumInfo _albumInfo;
         private readonly IArtistMetadataRepository _artistMetadataRepository;
         private readonly IRefreshTrackService _refreshTrackService;
         private readonly Logger _logger;
 
         public AddAlbumService(IAlbumService albumService,
+                               IReleaseService releaseService,
                                IProvideAlbumInfo albumInfo,
                                IArtistMetadataRepository artistMetadataRepository,
                                IRefreshTrackService refreshTrackService,
                                Logger logger)
         {
             _albumService = albumService;
+            _releaseService = releaseService;
             _albumInfo = albumInfo;
             _artistMetadataRepository = artistMetadataRepository;
             _refreshTrackService = refreshTrackService;
             _logger = logger;
         }
 
-        public Album AddAlbum(Album newAlbum)
+        public List<AlbumRelease> AddAlbumReleases(Album album)
         {
-            Ensure.That(newAlbum, () => newAlbum).IsNotNull();
-
-            var tuple = AddSkyhookData(newAlbum);
-            newAlbum = tuple.Item2;
+            var remoteReleases = album.AlbumReleases.Value.DistinctBy(m => m.ForeignReleaseId).ToList();
+            var existingReleases = _releaseService.GetReleasesForRefresh(album.Id, remoteReleases.Select(x => x.ForeignReleaseId));
```

Comment:
Would this ever find the old releases given they are linked to the previous album (release group) and this call is using the new albumId that was just created to find the `existingReleases`. Maybe you need a function in releaseService that only accepts a list of releases (releaseForeignIds) without the constraint of an album.

Something like `_releaseService.GetReleasesForRefresh(remoteReleases.Select(x => x.ForeignReleaseId))`.

Assuming the test only works because we are mocking with a response we want to see from the function above.